cli/src/deno.rs

```diff
@@ -199,35 +206,86 @@ impl ModuleLoader for ExtensionsModuleLoader {
             };
 
             if should_transpile {
-                code = transpile(module_specifier.to_string(), code, media_type)?;
+                let transpiled =
+                    transpiler.transpile(module_specifier.to_string(), code, media_type)?;
+                code = transpiled.text.clone();
             }
 
             Ok(ModuleSource::new(module_type, code.into(), &module_specifier))
         })
     }
 }
 
-/// Transpile code to JavaScript.
-fn transpile(
-    specifier: impl Into<String>,
-    code: impl Into<String>,
-    media_type: MediaType,
-) -> Result<String> {
-    let parsed = deno_ast::parse_module(ParseParams {
-        text_info: SourceTextInfo::from_string(code.into()),
-        specifier: specifier.into(),
-        capture_tokens: false,
-        scope_analysis: false,
-        maybe_syntax: None,
-        media_type,
-    })?;
-    Ok(parsed.transpile(&Default::default())?.text)
+impl SourceMapGetter for ExtensionsModuleLoader {
+    fn get_source_map(&self, file_name: &str) -> Option<Vec<u8>> {
+        let transpiled = self.transpiler.cache.get(file_name)?;
+        transpiled.source_map.clone().map(|map| map.into_bytes())
+    }
+
+    fn get_source_line(&self, file_name: &str, line_number: usize) -> Option<String> {
+        let source_map = self.get_source_map(file_name)?;
+
+        let text = String::from_utf8(source_map).ok()?;
+
+        text.lines().nth(line_number).map(|line| line.to_owned())
```

Comment:
This gets a particular line of the source map... Not the source itself. This leads to very weird errors:

```sh-session
> cat foo/main.ts
throw new Error();

> ./target/debug/phylum extension run -y ./foo
Extension error: Uncaught Error
{"version":3,"sources":["file:///Users/kyle/code/cli/foo/main.ts"],"sourcesContent":["throw new Error();\n"],"names":[],"mappings":"AAAA,MAAM,IAAI"}
      ^
    at file:///Users/kyle/code/cli/foo/main.ts:1:7
```